controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -127,6 +127,19 @@ func (r *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
 		)
 	}
 
+	if *r.Spec.Replicas <= 0 {
```

Comment:
Thanks for catching this. We should probably consolidate the common validation between create/update, similarly we probably also want to ensure r.Spec.Replicas is non-nil on Update as well.

Reply:
Is this a suggestion for this PR or is this a follow on suggestion?

Reply:
I'm currently working on a PR to @dlipovetsky's branch

Reply:
https://github.com/dlipovetsky/cluster-api/pull/2

Reply:
what about merging this PR and opening one against capi?

Reply:
:man_shrugging: either way is fine for me.